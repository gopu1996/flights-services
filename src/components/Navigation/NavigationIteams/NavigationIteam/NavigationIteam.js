import React from "react";

import { NavLink } from "react-router-dom";

const navigationIteam = (props) => (
  <li className="NavigationItem">
    <NavLink exact={props.exact} to={props.link} activeClassName="active">
      {props.children}
    </NavLink>
  </li>
);

export default navigationIteam;
