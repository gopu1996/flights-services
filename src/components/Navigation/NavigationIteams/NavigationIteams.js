import React from "react";
import "../../../sass/main.scss"
import NavigationIteam from "./NavigationIteam/NavigationIteam";

const NavigationIteams = (props) => (
  <ul className="NavigationItems">
    {!props.isAuth ? (
      <NavigationIteam link="/">LOGIN</NavigationIteam>
    ) : (
      <NavigationIteam link="/logout">LOGOUT</NavigationIteam>
    )}
  </ul>
);
export default NavigationIteams;
