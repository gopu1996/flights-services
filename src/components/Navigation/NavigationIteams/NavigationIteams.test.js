import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NavigationIteams from "./NavigationIteams";
import NavigationIteam from "./NavigationIteam/NavigationIteam";

configure({ adapter: new Adapter() });
let wrapper;
describe("<NavigationIteams />", () => {
  it('renders correctly', () => {
    wrapper = shallow(<NavigationIteams />);
    expect(wrapper).toMatchSnapshot();
});
  it("should render one <NavigationItem /> elements if not authenticated", () => {
    wrapper = shallow(<NavigationIteams />);
    expect(wrapper.find(NavigationIteam)).toHaveLength(1);
  });
  it("should render one <NavigationItem /> elements if authenticated", () => {
    // wrapper = shallow(<NavigationItems isAuthenticated />);
    wrapper.setProps({ isAuth: true });
    expect(wrapper.find(NavigationIteam)).toHaveLength(1);
  });

  it("should render one <NavigationItem /> element with Logout label if authenticated", () => {
    wrapper.setProps({ isAuth: true });
    expect(
      wrapper.contains(<NavigationIteam link="/logout">LOGOUT</NavigationIteam>)
    ).toEqual(true);
  });
});
