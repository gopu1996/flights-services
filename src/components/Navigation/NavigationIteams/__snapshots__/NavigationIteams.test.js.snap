// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationIteams /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationIteams />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <navigationIteam
        link="/"
>
        LOGIN
</navigationIteam>,
      "className": "NavigationItems",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "LOGIN",
        "link": "/",
      },
      "ref": null,
      "rendered": "LOGIN",
      "type": [Function],
    },
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <navigationIteam
          link="/"
>
          LOGIN
</navigationIteam>,
        "className": "NavigationItems",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "LOGIN",
          "link": "/",
        },
        "ref": null,
        "rendered": "LOGIN",
        "type": [Function],
      },
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
